### PR TITLE
IBX-8824: The popup for custom tags with multiple fields is not scrollable

### PR DESCRIPTION
**Pull request by lucasOsti**
| :ticket: Issue | IBX-8824|
|----------------|-----------|

### CODE REVIEW

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js [hunk 1]
```diff
@@ -217,6 +219,23 @@ class IbexaCustomTagUI extends Plugin {
         return { target: view.domConverter.viewRangeToDom(range) };
     }
 
+    setPanelContentMaxHeight() {
+        const { innerHeight: windowHeight } = window;
+        const { top: panelTopPosition } = this.balloon.view;
+        const { element: panelNode } = this.balloon.view;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js:225]:
```suggestion
        const { top: panelTopPosition, element: panelNode } = this.balloon.view;
```

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js [hunk 2]
```diff
@@ -217,6 +219,23 @@ class IbexaCustomTagUI extends Plugin {
...
+        const panelHeader = panelNode.querySelector('.ibexa-custom-tag-panel-header');
+        const panelContent = panelNode.querySelector('.ibexa-custom-tag-panel-content');
+        const panelFooter = panelNode.querySelector('.ibexa-custom-tag-panel-footer');
+        const panelHeaderHeight = panelHeader?.offsetHeight || 0;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js:229]:
```suggestion
        const panelHeaderHeight = panelHeader?.offsetHeight ?? 0;
```

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js [hunk 3]
```diff
@@ -217,6 +219,23 @@ class IbexaCustomTagUI extends Plugin {
...
+        const panelFooterHeight = panelFooter?.offsetHeight || 0;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js:230]:
```suggestion
        const panelFooterHeight = panelFooter?.offsetHeight ?? 0;
```

File: src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js [hunk 4]
```diff
@@ -115,6 +116,23 @@ class IbexaInlineCustomTagUI extends Plugin {
         return { target: view.domConverter.viewRangeToDom(range) };
     }
 
+    setPanelContentMaxHeight() {
+        const { innerHeight: windowHeight } = window;
+        const { top: panelTopPosition } = this.balloon.view;
+        const { element: panelNode } = this.balloon.view;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js:122]:
```suggestion
        const { top: panelTopPosition, element: panelNode } = this.balloon.view;
```

File: src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js [hunk 5]
```diff
@@ -115,6 +116,23 @@ class IbexaInlineCustomTagUI extends Plugin {
...
+        const panelHeader = panelNode.querySelector('.ibexa-custom-tag-panel-header');
+        const panelContent = panelNode.querySelector('.ibexa-custom-tag-panel-content');
+        const panelFooter = panelNode.querySelector('.ibexa-custom-tag-panel-footer');
+        const panelHeaderHeight = panelHeader?.offsetHeight || 0;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js:126]:
```suggestion
        const panelHeaderHeight = panelHeader?.offsetHeight ?? 0;
```

File: src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js [hunk 6]
```diff
@@ -115,6 +116,23 @@ class IbexaInlineCustomTagUI extends Plugin {
...
+        const panelFooterHeight = panelFooter?.offsetHeight || 0;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/inline-custom-tag/inline-custom-tag-ui.js:127]:
```suggestion
        const panelFooterHeight = panelFooter?.offsetHeight ?? 0;
```

File: src/bundle/Resources/public/scss/_balloon-form.scss [hunk 7]
```diff
@@ -1,16 +1,18 @@
...
             border-top-right-radius: $ibexa-border-radius;
-            padding: calculateRem(2px) 0;
+            padding: calculateRem(2px) calculateRem(16px);
             font-weight: bold;
         }
 
         &__fields {
-            padding: calculateRem(8px) 0;
+            overflow: auto;
+
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/scss/_balloon-form.scss:14]:
```suggestion
```

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

**Review by tischsoic — APPROVED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Review by barbaragr — CHANGES_REQUESTED** [review]
Hi, I've found an issue:
1. Go to creating an article
2. Paste some text into richtext field
3. Somewhere in the middle insert added custom tag
4. Don't fill the fields in CT, just click save
5. Click settings button and the edit button in custom tag
6. Try to scroll to the top 
**Actual result:** Top of the custom tag is still covered. See attachment, please.
https://github.com/user-attachments/assets/a77eaf4e-016c-4a3e-829b-5dad652fd586

**Comment by alongosz** [comment]
Note: 4.6 CI fix is pending review, via #184

**Review by tischsoic — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Comment by barbaragr** [comment]
Still covers the very last element in menu when it's displayed below custom tag... 😅 

https://github.com/user-attachments/assets/3287b024-315c-4fa8-9f5f-55db607d904c

**Review by tischsoic — APPROVED** [review]

**Review by tischsoic — APPROVED** [review]

**Review by KamilSznajdrowicz — CHANGES_REQUESTED** [review]
Hi, I've found an issue:

1. Go to creating an article
2. Paste some text into richtext field
3. Enter distraction-free mode.
4. Somewhere in the middle insert added custom tag
5. Try to scroll to the top
Actual result: A window with a scrollbar appears in the wrong place. See attachment, please.

https://github.com/user-attachments/assets/49e229ed-fb5b-4975-81ac-37de3d61715e

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js [hunk 8]
```diff
@@ -19,6 +19,22 @@ class IbexaCustomTagUI extends Plugin {
         this.addCustomTag = this.addCustomTag.bind(this);
 
         this.isNew = false;
+        this.lastClassListValue = null;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js:22]: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#attributeoldvalue
there is dedicated option for it in mutation observer :)

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js (hunk 8 above)
Comment (lucasOsti) [reply src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-ui.js:22]: I'm using `mutation.target.classList` and this object don't have `oldValue` property

**Review by KamilSznajdrowicz — CHANGES_REQUESTED** [review]
Hi, I've found new an issue:
The popup disappears when we start scrolling the page up and down quickly. The error occurs in both regular and distraction-free mode.
1. Go to creating an article
2. Paste some text into richtext field
3. Somewhere in the middle insert added custom tag
4. Scroll the page up and down quickly.
Please see the attachment:
https://github.com/user-attachments/assets/1d7df575-1bfd-45fd-b2f2-605407d5e233

**Review by barbaragr — APPROVED** [review]

**Review by KamilSznajdrowicz — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Comment by tischsoic** [comment]
Merged up:
- main: https://github.com/ibexa/fieldtype-richtext/commit/c2d81683a3a4b568ae32aefabd37155aa8e1cec1

**Pull request merged**